### PR TITLE
Notifications: Mention enhancements

### PR DESCRIPTION
**Pull request by brianp**
## Spec
I want to see recognizable avatars beside users names/emails when they have been mentioned in comments.
I want the mentions pop up to continue working as I navigate through the application.
I want the mentions pop up to disappear when I scroll instead of hovering absolutely on the page.

Currently when a user is mentioned in a comment their email will show up as @user@company.com In the comment. Instead we want it to show the name (pro), or email (ce) alongside their avatar.

## Proposed solution
Update the comment formatters to provide the avatars and present the username/email better. Fix the scroll bug when mentions are opened. Fix the bug that prevents mentions from working.

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### Check List

- [x] Added a CHANGELOG entry

### CODE REVIEW

File: app/controllers/concerns/commented.rb [hunk 1]
```diff
@@ -2,14 +2,36 @@ module Commented
...
+  def mentioned_users(comments)
+    # Match any non whitespace that starts with an @ has another @
+    emails = comments.inject([]) do |collection, comment|
+      (collection + comment.scan(/@(\S*@\S*)/)).flatten.uniq
+    end
+
+    current_project.authors.where(email: emails)
+  end
+
+  def mentions_builder(comments)
```
Comment (brianp) [inline app/controllers/concerns/commented.rb]: I'd really like to write some test code for this with some email edge cases etc but the codebase doesn't do controller specs, let alone concerns for controllers. If anyone has a good idea where a place for testing this could go let me know and I'd cook something up.

File: app/assets/javascripts/snowcrash/modules/mentions.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      tribute = new Tribute({
+        allowSpaces: function() { return false },
+        menuItemTemplate: function(item) {
+          return '<img src="' + item.original.avatar_url + '" width="24px" height="24px" > ' + item.string
+        },
+        noMatchTemplate: function() { return '' },
+        values: JSON.parse($('meta[name=mentionable-users]').attr('content'))
+      });
+
+      $('[data-behavior=mentions-scroll]').scroll(function(){
```
Comment (etdsoft) [inline app/assets/javascripts/snowcrash/modules/mentions.js]:
```suggestion
      $('[data-behavior~=mentions-scroll]').scroll(function(){
```
This allows for multiple behavior stacking on a single element

File: app/controllers/activities_controller.rb [hunk 3]
```diff
@@ -3,6 +3,7 @@
 
 class ActivitiesController < AuthenticatedController
   include ProjectScoped
+  include Commented
```
Comment (etdsoft) [inline app/controllers/activities_controller.rb]: :alpha

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 class CommentsController < AuthenticatedController
   include ActivityTracking
   include ProjectScoped
+  include Commented
```
Comment (etdsoft) [inline app/controllers/comments_controller.rb]: :aplha

File: app/presenters/activity_presenter.rb [hunk 5]
```diff
@@ -88,7 +88,8 @@ def avatar_image(size)
         class: 'gravatar',
         data: { fallback_image: image_path('logo_small.png') },
         title: activity.user,
-        width: size
+        width: size,
+        style: "width: #{size}px; height: #{size}px"
```
Comment (etdsoft) [inline app/presenters/activity_presenter.rb]: :alpha

File: app/presenters/notification_presenter.rb [hunk 6]
```diff
@@ -59,7 +59,8 @@ def avatar_image(size)
         class: 'gravatar',
         data: { fallback_image: image_path('logo_small.png') },
         title: notification.actor.email,
-        width: size
+        width: size,
+        style: "width: #{size}px; height: #{size}px"
```
Comment (etdsoft) [inline app/presenters/notification_presenter.rb]: :alpha
